File: Algorithms/Mathematics/Factorial/Interative/java/Factorial.java
import java.util.Scanner;

public class Factorial {
    public static long factorial(int n) {
        long fact = 1;
        if(n > 1)
            for(int i=2; i<=n; i++)
                fact *= i;
        return fact;
    }

    public static void main(String[] args) {
        System.out.print("Enter the number: ");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        sc.close();
        System.out.println("Factorial of " + n + " is: " + factorial(n));
    }
}

File: Algorithms/Mathematics/Factorial/Recursive/python/factorial.py
import sys
sys.setrecursionlimit(1500)

def factorial(n):
    fact = n
    if n < 2:
        return 1
    else:
        return fact * factorial(n-1)

if __name__ == '__main__':
    n = int(input("Enter the number: "))
    print(f"Factorial of {n} is {factorial(n)}")
    